src/io/stakater/pods/Pod.groovy

```diff
@@ -6,17 +6,21 @@ def setToolsImage(Map parameters = [:], String image) {
     parameters.podContainers.defaultContainer.image = image
 }
 
-def mountDockerSocket(Map parameters = [:]) {
-    setPodVolumes(parameters)
-    parameters.podVolumes.isDockerMount = true
-}
-
 Map setDockerConfig(Map parameters = [:]) {
-    mountDockerSocket(parameters)
+    setPodVolumes(parameters)
     setDefaultContainerEnvVarsConfig(parameters)
 
-    parameters.podVolumes.isDockerConfig = true
-    parameters.podContainers.defaultContainer.envVarsConfig.isDocker = true
+    // Don't mount docker socket if useBuildah is true or isDockerMount is true
+    if (!((parameters.containsKey("useBuildah") && parameters.get('useBuildah', true )) ||
+            parameters.containsKey("isDockerMount") && parameters.get('isDockerMount', false ))) {
+        parameters.podVolumes.isDockerMount = true
+        parameters.podContainers.defaultContainer.envVarsConfig.isDocker = true
+    }
+
+    // Don't mount docker config if isDockerConfig is false
+    if (!(parameters.containsKey("isDockerConfig") && parameters.get('isDockerConfig', false ))) {
```

Comment:
Does the condition need to be inverted here? Seems like if the key doesn't exist or if value of `parameters. isDockerConfig` is false then `parameters.podVolumes.isDockerConfig = true` would be set to true?

Reply:
If key exists and key is set to false, then isDockerConfig will use the default value false. Else isDockerConfig should be set to true.

Reply:
Handled